### ISSUE TITLE
Upload failed only on Webfrontend: Upload Max. 0 B - Not enough storage available

### ISSUE THREAD

**Issue by DavidWiesner** [owncloud/core#15030]
Hi,

thanks for this great project!

I have an issue uploading files on my Webfrontend, while its working with a sync client.

### Steps to reproduce on some installations
 1. There is enough free space on the disk (but owncloud can´t detect and will return `\OCP\Files\FileInfo::SPACE_UNKNOWN`)
 2. Tooltip next to the upload button says "Upload Max. 0 B". 
 3. If I try to upload a file anyway the file progressbar indicates that a file will be uploaded. 
 4. If the progessbar is on 100%, i got an error message "Not enough storage available".

### Expected behaviour
an upload shouldn´t fail if the freedisk space is unkown.

### Actual behaviour
the upload failed

### The Problem
In the end I figure out its seems a PHP problem on some systems. But owncloud should work anyway!

The root of the evil is `INF == 0` on some installations. Here is the bugreport https://bugs.php.net/bug.php?id=69188 
 
Since you are using INF in `freeSpace()` in [`lib/private/helpers.php#L824`](https://github.com/owncloud/core/blob/06aef4e8b1f2086971c62b064f810e24785fb215/lib/private/helper.php#L824)

`maxUploadFilesize()` using the result of `freeSpace()` in a `min()` function (see: [lib/private/helper.php#L815](https://github.com/owncloud/core/blob/06aef4e8b1f2086971c62b064f810e24785fb215/lib/private/helper.php#L815) ). So `maxUploadFilesize()` will always return 0.

In the end the fileupload will be canceled  by a if-statement in [apps/files/ajax/upload.php#L108](https://github.com/owncloud/core/blob/06aef4e8b1f2086971c62b064f810e24785fb215/apps/files/ajax/upload.php#L108)

### Possible Fix

Using `PHP_INT_MAX` instead of `INF` in my `lib/private/helpers.php` fixed the problem on my installation. 
```
--- old/helper.php	2015-03-19 12:13:30.818505473 +0100
+++ new/helper.php	2015-03-19 12:11:21.924501375 +0100
@@ -831,7 +831,7 @@
 			$freeSpace = max($freeSpace, 0);
 			return $freeSpace;
 		} else {
-			return INF;
+			return PHP_INT_MAX;
 		}
 	}
```
regards,
David

### Server configuration
**Operating system**:
SunOS  5.10  i86pc
**Web server:**
Apache/2.2.29
**Database:**
MySQL 5.0.96
**PHP version:**
5.6.6
**ownCloud version:** (see ownCloud admin page)
8.0.2 (same problem in 8.0, 7.0.5)
**Updated from an older ownCloud or fresh install:**
updated
**List of activated apps:**
files, pictures, calendar, contacts
**The content of config/config.php:**
```
<?php
$CONFIG = array (
  'instanceid' => /*...*/,
  'passwordsalt' => /*...*/,
  'datadirectory' => /*...*/,
  'dbtype' => 'mysql',
  'version' => '8.0.2.0',
  'dbname' => /*...*/,
  'dbhost' => /*...*/,
  'dbtableprefix' => /*...*/,
  'dbuser' => /*...*/,
  'dbpassword' => /*...*/,
  'installed' => true,
  'maintenance' => false,
  'theme' => '',
  'loglevel' => '0',
  'trusted_domains' => 
  array (/*...*/
  ),
  'secret' => /*...*/,
  'share_folder' => '/Shared',
);
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
local
**Are you using encryption:** yes/no
no
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no
### Client configuration
**Browser:**
Chrome/Firefox/Opera
**Operating system:**
Linux
### Logs
no corresponding logs found

**Comment by DeepDiver1975** [comment]
@icewind1991 does that make sense?

**Comment by karlitschek** [comment]
@icewind1991 @DeepDiver1975 What is the status here?

**Comment by PVince81** [comment]
> The root of the evil is INF == 0 on some installations. Here is the bugreport https://bugs.php.net/bug.php?id=69188

lol wut ?

Replacing INF with another value makes sense, not sure about PHP_INT_MAX though.

**Comment by icewind1991** [comment]
int_max seems fine to me

**Comment by cmonteroluque** [comment]
This bug will fall off 8.2 if not resolved by 6 Oct

**Comment by icewind1991** [comment]
Fix is here: #19533
